cuqi/experimental/mcmc/_langevin_algorithm.py

```diff
@@ -225,3 +249,140 @@ def _log_proposal(self, theta_star, theta_k, g_logpi_k):
+
+    target : `cuqi.distribution.Distribution`
+        The target distribution to sample from. The target distribution result from
+        a differentiable likelihood and prior of type RestorationPrior.
+    
+    initial_point : ndarray
+        Initial parameters. *Optional*
+
+    scale : float
+        The Langevin diffusion discretization time step (In practice, scale must
+        be smaller than 1/L, where L is the Lipschitz of the gradient of the log
+        target density, logd).
+        
+    smoothing_strength : float
+        This parameter controls the smoothing strength of MYULA. smoothing_strength
+        must be equal to restoration_strength of the RestoratioPrior.
+
+    callback : callable, *Optional*
+        If set this function will be called after every sample.
+        The signature of the callback function is `callback(sample, sample_index)`,
+        where `sample` is the current sample and `sample_index` is the index of
+        the sample.
+        An example is shown in demos/demo31_callback.py.
+
+    A Deblur example can be found in demos/howtos/myula.py
+    # TODO: update demo once sampler merged
+    """
+    def __init__(self, target=None, scale=1.0, smoothing_strength=0.1, **kwargs):
+        self.smoothing_strength = smoothing_strength
+        super().__init__(target=target, scale=scale, **kwargs)
+    
+    @SamplerNew.target.setter
+    def target(self, value):
+        """ Set the target density. Runs validation of the target. """
+        self._target = value
+
+        if self._target is not None:
+            # Create a smoothed target
+            copied_value = deepcopy(value)
+            if isinstance(copied_value.prior, RestorationPrior):
+                copied_value.prior = MoreauYoshidaPrior(
+                    copied_value.prior,
+                    self.smoothing_strength,
+                    geometry=copied_value.prior.geometry)
+            self._smoothed_target = copied_value
+
+            # Validate the target
+            self.validate_target()
+            
+    def validate_target(self):
+        # Call ULANew target validation
+        super().validate_target()
+
+        # Additional validation for MYULA target
+        if isinstance(self.target.prior, MoreauYoshidaPrior):
+            raise ValueError(("The prior is already smoothed, apply"
+                              " ULA when using a MoreauYoshidaPrior."))
+        if not isinstance(self.target.prior, RestorationPrior):
+            raise NotImplementedError(("Using MYULA with other than"
+                                       " RestorationPrior prior"
+                                       " is not implemented yet."))
+
+    def _eval_target_grad(self, x):
+        if isinstance(self.target.prior, cuqi.implicitprior.RestorationPrior) \
+        and not np.isclose(self.smoothing_strength, self.target.prior.restoration_strength):
+            raise ValueError(f"smoothing_strength of the {self.__class__.__name__}"
```

Comment:
Same comment as above, I wonder if it is possible to modify this check or design the interface different such that it is either not possible to pass a different value, or it is not required to pass the same value.

Reply:
Thank you for your comment Jakob! That is a good point. We updated the constructor of RestorationPrior and removed passing restoration_strength to it as we discussed in the meeting from 08/10/2024.
Now, RestorationPrior does not contain this attribute anymore and there is no need to enforce this equality. So this warning was removed.